client_handlers.go

```diff
@@ -0,0 +1,127 @@
+package irc
+
+import (
+	"fmt"
+	"strings"
+)
+
+type clientFilter func(*Client, *Message)
+
+// clientFilters are pre-processing which happens for certain message
+// types. These were moved from below to keep the complexity of each
+// component down.
+var clientFilters = map[string]clientFilter{
+	"001":  handle001,
+	"433":  handle433,
+	"437":  handle437,
+	"PING": handlePing,
+	"PONG": handlePong,
+	"NICK": handleNick,
+	"CAP":  handleCap,
+}
+
+func handle001(c *Client, m *Message) {
```

Comment:
can you add human names to these codes?

Reply:
I coped the descriptions from the relevant RFC, along with the given names.